model/manifest.go

```diff
@@ -198,7 +198,7 @@ type Manifest struct {
 type ManifestServer struct {
 	// Executables are the paths to your executable binaries, specifying multiple entry points
 	// for different platforms when bundled together in a single plugin.
-	Executables *ManifestExecutables `json:"executables,omitempty" yaml:"executables,omitempty"`
+	Executables map[string]string `json:"executables,omitempty" yaml:"executables,omitempty"`
```

Comment:
Ah, this is something I did not anticipate: we apply SemVer to our `model` package, and this is unfortunately a backwards incompatible change to the signature of this model object.

To preserve SemVer, could we customize the deserialization to both populate `Executables` and a new, private field (`otherExecutables map[string]string`?) that is used within `GetExecutableForRuntime` to ultimately achieve the same effect?

Reply:
It appears that Go doesn't have a mechanism to handle "unknown fields" via its `Encoder` nor `Unmarshal` for both `encoding/json` and `gopkg.in/yaml.v2`. We could add special logic inside `FindManifest` which will handle this specific case but I'm not sure if that's a good idea and it feels really hacky.

To retain values of "unknown fields", the simplest way is to unmarshal/decode it into a generic `map[string]interface` and take the values from there and apply it to the created struct variable.

Another issue with this approach is when you want to encode/marshal it into a json/yaml file. `encoding/json` and `gopkg.in/yaml.v2` doesn't have anything to manipulate json/yaml data directly and would mean a special function to just handle encoding for that part.

Reply:
Thanks Muhammad! @lieut-data  can you take a look? thanks!

Reply:
Thanks @yukiisbored! Preserving SemVer is a hard requirement until the next major version bump. Could we define an unexported `manifestServer` of this `struct` with the desired `map[string]string`, define `UnmarshalJSON` on `ManifestServer` to unmarshal to an instance of `manifestServer` and then copy over the fields in question, including to the new `otherExecutables`?

Reply:
Due to how [`reflect`](https://golang.org/pkg/reflect/) works in Go, `encoding/json` can only work if the field is exported. It won't handle unexported fields: https://play.golang.org/p/uoXRQFwmesY

Even if we decide to export that field, it would result in another issue: You cannot have multiple fields with the same json tag. Doing so it would result in both fields being ignored: https://play.golang.org/p/5KPU9OuDsP1

I also tried experimenting with embedding structs which might allow us to have two versions of `Manifest` or `ManifestServer` but it ended up with the same roadblock as before: https://play.golang.org/p/ofeuXSTvdzH

 I'm not sure if there's a way to do this without importing another library just to handle this specific case or have a separate copy of `Manifest` which uses the "new" `ManifestServer` structure. 

Another point to consider is the fact that I don't see `ManifestServer` being used directly anywhere outside of `manifest.go`, so I'm not sure if introducing a change which would break how plugin manifest data is handled to be worth it.

Reply:
On the "importing another library" case, we could import a library like [jsonparser](https://github.com/buger/jsonparser) or [fastjson](https://github.com/valyala/fastjson) which allows parsing JSON without a schema.

With it we can go to the original unexported fields route, I'm just unsure whether it's a good idea to import another library just for this case and how the call to `GetExecutableForRuntime` might look like since the original JSON data is required for this to work, might need to see how the manifest itself is loaded for this very specific case.